tasks/main.yml

```diff
@@ -0,0 +1,97 @@
+---
+- name: Install "npm"
+  apt:
+    name: npm
+    update_cache: yes
+  with_items:
+    - npm
+    - nodejs-legacy
+    - nodejs
+
+- name: Create hubot install directory
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ mattermost_hubot_user }}"
+    group: "{{ mattermost_hubot_group }}"
+  with_items:
+    - "{{ mattermost_hubot_install_path }}"
+    - "{{ mattermost_hubot_install_path }}/.npm"
+
+# We are using shell and sudo since npm does not respect the user under which a process should run
+- name: Install "yo" and "generator-hubot" node.js package
+  environment:
+    HOME: "{{ mattermost_hubot_install_path }}"
+    npm_config_user: "{{ mattermost_hubot_user }}"
+    npm_config_init_module: "{{ mattermost_hubot_install_path }}/.npm-init.js"
+    npm_config_userconfig: "{{ mattermost_hubot_install_path }}/.npmrc"
+    npm_config_cache: "{{ mattermost_hubot_install_path }}/.npm"
+  shell: "sudo -E -u {{ mattermost_hubot_user }} npm install {{ item }}"
+  changed_when: False
+  args:
+    chdir: "{{ mattermost_hubot_install_path }}"
+  with_items:
+    - yo
+    - generator-hubot
+
+- name: Init hubot
+  become: True
+  shell: "sudo -u {{ mattermost_hubot_user }} -E yo hubot --adapter matteruser --owner {{ mattermost_hubot_owner }} --name {{ mattermost_hubot_name }} --description '{{ mattermost_hubot_description }}'"
+  environment:
```

Comment:
Can you extract the environment variables to vars and only define them once?

Reply:
fixed